azerothcore/azerothcore-wotlk
# chore(Core/Condition): replace assertion with logs preventing server crash

Ignore the commit with code-style, check only what I changed, it's really a small change.
https://github.com/azerothcore/azerothcore-wotlk/pull/7730/commits/facaf3ddab2f31f815aaf19374ddd820e6b89cba

## Changes Proposed:
-  replace ASSERT with logs

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7577

## Tests Performed:
- I didn't test it, but it should not break anything

## How to Test the Changes:
I suggest to import this on your private server and check if this crash happens again.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
